Partner Pawn Job Training Bonus Setting

Nirrudn-SOK opened the pull request:
* Adds extra credit on Job Training kills when the player's partner pawn is present.

Based on a changelog/Tweet from April 3rd, 2018, and exposed as a game setting for server admins to change/disable as desired.

# Checklist:
- [x] The project compiles
- [x] The PR targets `develop` branch

Nirrudn-SOK committed:
Add Partner Pawn Bonus to Job Training

Adds a bonus kill count to job training enemies if the player's partner pawn is in the party, based on an official Tweet on April 3rd, 2018.

Nirrudn-SOK committed:
Merge branch 'sebastian-heinz:develop' into feat/partner-pawn-job-training

Nirrudn-SOK committed:
Add Partner Pawn Job Training Bonus to Game Settings

Exposed the behavior as a game setting so it can be tweaked/disabled as desired by server admins.

Nirrudn-SOK committed:
Fixed Default

Switched from a multiplier to an added bonus for clarity, forgot to set the default case to 0, oopsy.

pacampbell commented on Arrowgene.Ddon.GameServer/Characters/JobMasterManager.cs:
```diff
@@ -55,6 +56,10 @@ public PacketQueue HandleEnemyKill(GameClient client, InstancedEnemy enemy, DbCo
             }
 
+            //Kill count bonus for having a partner pawn present
+            uint partnerKillBonus = (uint)((client.Character.PartnerPawnId != 0 &&
```
Can we add some of this logic to the partner pawn manager?

```c#
public bool IsPartnerPawnInParty(Client client)
{
    return client.Party.Members.Where(x => x.IsPawn).Any(x => x.PawnId == client.Character.PartnerPawnId);
}
```

Then in this job master function
```c#
uint partnerKillBonus = IsPartnerPawnInParty(client) ? Server.GameSettings.GameServerSettings.JobTrainingPartnerBonus : 0;
```

pacampbell reviewed (commented):
Thanks for making the change. Let's add the logic to the partner pawn manager to prevent leakage of finding the partner pawn in the party (in case it changes).

Nirrudn-SOK committed:
Move logic to PartnerPawnManager

Per feedback, moved the check logic to a new function in the Partner Pawn Manager.

Nirrudn-SOK committed:
Merge branch 'develop' into feat/partner-pawn-job-training
